app_tests/integration_tests/llm/shortfin/tracing_test.py

```diff
@@ -0,0 +1,153 @@
+"""
+Integration test to ensure tracing functionality works correctly with the Shortfin LLM server.
+This tests that the server can generate a valid tracy capture file without crashing.
+"""
+
+import logging
+import os
+import pytest
+import requests
+import time
+import uuid
+import subprocess
+from pathlib import Path
+from typing import Dict, Any, Tuple
+
+from ..model_management import AccuracyValidationException
+from ..server_management import ServerInstance, ServerConfig
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.parametrize(
+    "model_artifacts,server",
+    [
+        ["tinystories_llama2_25m", {"prefix_sharing": "none"}],
+    ],
+    indirect=True,
+)
+
+# Same golden values as in tinystories_llama2_25m_test.py
+GOLDEN_PROMPT = "Once upon a time"
+GOLDEN_RESPONSE = ", there was a little girl named Lily."  # this assumes purely deterministic greedy search
+
+
+class TestTracingFunctionality:
+    """Test suite for tracing functionality with the LLM server."""
+
+    def test_tracing_generation(self, server: Tuple[Any, int], tmp_path: Path) -> None:
+        """Tests that tracing works while running text generation.
+
+        Args:
+            server: Tuple of (process, port) from server fixture
+            tmp_path: Temporary directory for storing the trace file
+        """
+        process, port = server
+        assert process.poll() is None, "Server process terminated unexpectedly"
+
+        # Determine output directory for artifacts (use environment variable if set, otherwise tmp_path)
+        output_dir = os.environ.get("TEST_OUTPUT_DIR")
+        if output_dir:
+            output_dir = Path(output_dir)
+            output_dir.mkdir(parents=True, exist_ok=True)
+        else:
+            output_dir = tmp_path
+
+        # Start tracy capture
+        capture_file = output_dir / "capture.tracy"
+        logger.info(f"Tracy capture will be saved to: {capture_file}")
+
+        tracy_process = subprocess.Popen(
+            ["iree-tracy-capture", "-o", str(capture_file)],
```

Comment:
This assumes that `iree-tracy-capture` exists, which is not true on Windows and macOS at the moment. Could skip the test on those platforms or if the tool does not appear on `$PATH`. This issue tracks building on all platforms and bundling for consistency: https://github.com/iree-org/iree/issues/19841